libcontainer/factory_linux.go

```diff
@@ -378,6 +379,31 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 		return fmt.Errorf("unable to convert _LIBCONTAINER_LOGPIPE: %w", err)
 	}
 
+	// Get mount files (O_PATH).
+	// Keep mountFds as nil if there are no fds to mount. Note we assume (and verify!) this in other parts of the code.
+	var mountFds []*int
```

Comment:
Perhaps move the whole thing out to a separate function, for clarity.

```go
func parseMountFds() ([]int, error) {
    env := os.Getenv("_LIBCONTAINER_MOUNT_FDS")
    if env == "" {
        return nil, nil
    }
    .....
}

...

mountFds, err := parseMountFds()
if err != nil {
    return err
}
....

Reply:
Yeap, nice cleanup. I think that function would benefit of doing the same for other parts too (out of scope for this PR, of course). 

Fixed, thanks!